src/streaming/controllers/MediaController.js

```diff
@@ -87,10 +88,10 @@ function MediaController() {
         }
 
         if (tracks.length === 0) {
-            setTrack(selectInitialTrack(type, tracksForType), true);
+            setTrack(this.selectInitialTrack(type, tracksForType), true);
```

Comment:
What was the reason to add "this" here?

Reply:
I added `selectInitialTrack` to the instance props / called using `this` so that it can be overridden using the factory extend system. I think it would be very useful to allow library users to provide their own selection logic. If such a facility existed I probably would have used it instead of creating this PR. :grin: